src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -700,6 +705,296 @@ describe('Transactions Controller (Unit)', () => {
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chainId}/transactions/${safeAddress}/preview`)
+        .send(previewTransactionDto)
+        .expect(200)
+        .expect({
+          txInfo: {
+            type: 'Custom',
+            to: {
+              value: previewTransactionDto.to,
+              name: null,
+              logoUri: null,
+            },
+            dataSize: '16',
+            value: previewTransactionDto.value,
+            methodName: dataDecodedResponse.method,
+            actionCount: null,
+            isCancellation: false,
+          },
+          txData: {
+            hexData: previewTransactionDto.data,
+            dataDecoded: dataDecodedResponse,
+            to: {
+              value: previewTransactionDto.to,
+              name: null,
+              logoUri: null,
+            },
+            value: previewTransactionDto.value,
+            operation: previewTransactionDto.operation,
+            trustedDelegateCallTarget: null,
+            addressInfoIndex: {},
+          },
+        });
+    });
+
+    it('should preview a transaction even if the data cannot be decoded', async () => {
+      const previewTransactionDto = previewTransactionDtoBuilder()
+        .with('operation', CALL_OPERATION)
+        .build();
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const safeResponse = safeBuilder().with('address', safeAddress).build();
+      const chainResponse = chainBuilder().build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrl = `${safeConfigApiUrl}/api/v1/chains/${chainId}`;
+        const getSafeUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}`;
+        const getContractUrlPattern = `${chainResponse.transactionService}/api/v1/contracts/`;
+        if (url === getChainUrl) {
+          return Promise.resolve({ data: chainResponse });
+        }
+        if (url === getSafeUrl) {
+          return Promise.resolve({ data: safeResponse });
+        }
+        if (url.includes(getContractUrlPattern)) {
+          return Promise.reject({ status: 404 });
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
+      mockNetworkService.post.mockImplementation((url) => {
+        const getDataDecodedUrl = `${chainResponse.transactionService}/api/v1/data-decoder/`;
+        if (url === getDataDecodedUrl) {
+          return Promise.reject({ error: 'Data cannot be decoded' });
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chainId}/transactions/${safeAddress}/preview`)
+        .send(previewTransactionDto)
+        .expect(200)
+        .expect(({ body }) => {
+          expect(body).toEqual({
```

Comment:
`toEqual` doesn't seem required. We can test the `body` directly right?

Reply:
Right! Thanks. Changed in [9565c0a](https://github.com/5afe/safe-client-gateway-nest/pull/291/commits/9565c0ad8b6c6533745567f36f005f4915859bf4)